content/en/docs/concepts/scheduling-eviction/pod-eviction.md

```diff
@@ -0,0 +1,417 @@
+of another pod's resource consumption. If a system daemon (such as `kubelet`,
+`docker`, and `journald`) is consuming more resources than were reserved via 
+`system-reserved` or `kube-reserved` allocations, and the node only has
+`Guaranteed` or `Burstable` pods using less resources than requests left on it,
+then the kubelet must choose to evict one of these pods to preserve node stability
+and to limit the impact of resource starvation on other pods. In this case, it
+will choose to evict pods of lowest Priority first.
+
+When the kubelet evicts pods in response to `inode` or `PID` starvation, it uses
+the Priority to determine the eviction order, because `inodes` and `PIDs` have no
+requests.
+
+The kubelet sorts pods differently based on whether the node has a dedicated
+`imagefs` filesystem:
+
+#### With `imagefs`
+
+If `nodefs` is triggering evictions, the kubelet sorts pods based on `nodefs`
+usage (`local volumes + logs of all containers`).
+
+If `imagefs` is triggering evictions, the kubelet sorts pods based on the
+writable layer usage of all containers.
+
+#### Without `imagefs`
+
+If `nodefs` is triggering evictions, the kubelet sorts pods based on their total
+disk usage (`local volumes + logs & writable layer of all containers`)
+
+### Minimum eviction reclaim
+
+In some cases, pod eviction only reclaims a small amount of the starved resource.
+This can lead to the kubelet repeatedly hitting the configured eviction thresholds
+and triggering multiple evictions. 
+
+You can use the `--eviction-minimum-reclaim` flag or a [kubelet config file](/docs/tasks/administer-cluster/kubelet-config-file/)
+to configure a minimum reclaim amount for each resource. When the kubelet notices
+that a resource is starved, it continues to reclaim that resource until it
+reclaims the quantity you specify. 
+
+For example, the following configuration sets minimum reclaim amounts: 
+
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+evictionHard:
+  memory.available: "500Mi"
+  nodefs.available: "1Gi"
+  imagefs.available: "100Gi"
+evictionMinimumReclaim:
+  memory.available: "0Mi"
+  nodefs.available: "500Mi"
+  imagefs.available: "2Gi"
+```
+
+In this example, if the `nodefs.available` signal meets the eviction threshold,
+the kubelet reclaims the resource until the signal reaches the threshold of `1Gi`,
+and then continues to reclaim the minimum amount of `500Mi` it until the signal
+reaches `1.5Gi`. 
+
+Similarly, the kubelet reclaims the `imagefs` resource until the `imagefs.available`
+signal reaches `102Gi`. 
+
+The default `eviction-minimum-reclaim` is `0` for all resources.
+
+### Node out of memory behavior
+
+If the node experiences an out of memory (OOM) event prior to the kubelet
+being able to reclaim memory, the node depends on the [oom_killer](https://lwn.net/Articles/391222/)
+to respond.
+
+The kubelet sets an `oom_score_adj` value for each container based on the QoS for the pod.
```

Comment:
Might be worth to note somewhere that in addition to logic here if the pod has priority `system-node-critical` there is special case of it getting -997 score: 

https://github.com/kubernetes/kubernetes/blob/b58a7e2/pkg/kubelet/qos/policy.go#L41-L44

Reply:
(that could come in a follow-up PR, BTW)

Reply:
Lol, yes, I am going to make new suggestions in new PRs (but I implemented this one already).